agent/config/runtime.go

```diff
@@ -314,6 +314,16 @@ type RuntimeConfig struct {
 	// flag: -recursor string [-recursor string]
 	DNSRecursors []string
 
+	// DNSUseCache wether or not to use cache for dns queries
+	//
+	// hcl: dns_config { use_cache = (true|false) }
+	DNSUseCache bool
+
+	// DNSUseCache wether or not to use cache for dns queries
```

Comment:
Comment is same as previous option. Does it make sense to have DNSCacheMaxAge to 0 with DNSUseCache = true?

If not, I suggest to keep only DNSCacheMaxAge and enable cache if value > 0

Reply:
Having these two settings makes sense because results from endpoints that support blocking queries are never stale since a blocking refresh is running in background. 
This allows to benefit from the agent cache without sacrificing on staleness.
If however you are willing to receive stale results you can use `DNSCacheMaxAge` to cache the results from non-blocking endpoints such as prepared queries as well as continue using the results from blocking endpoints when the server connection is lost.